src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const result = [];
+
+  actions.forEach((elem) => {
+    const { type, extraData, keysToRemove } = elem;
+
+    switch (type) {
+      case 'addProperties':
+        copyState = Object.assign(copyState, extraData);
+
+        result.push({ ...copyState });
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete copyState[key];
+        }
+
+        result.push({ ...copyState });
+        break;
+
+      case 'clear':
+        copyState = {};
```

Comment:
Resetting `copyState` to an empty object is appropriate for the 'clear' action. This ensures that all properties are removed.